### PR TITLE
refactor(storage): adjust storage structure to add backend api

### PR DESCRIPTION
**Pull request by artek-koltun**
Sort out storage files to corresponding dirs since the number of file is growing, and it is getting harder to comprehend their structure. It was decided to stick to existing storage API separation for `frontend`/`middlend`/`backend`.
The directory structure in `cmd` mirrors the structure in the root `storage` libaray package to make it easier to match the related code.
Added `frontend` cmdline option in a separate commit to prepare to adding `backend` nvme part.

The `inventory`, `ipsec`, `evpn` code from `cmd` will be moved to corresponding dirs in the follow-up PRs.

Exemplary cmd set
```bash
ss0=$(/dpu create frontend nvme subsystem --nqn nqn.2019-06.io.spdk:opicli0 --hostnqn nqn.2019-06.io.spdk:opi-host0 --id subsys0)
ss1=$(/dpu create frontend nvme subsystem --nqn nqn.2019-06.io.spdk:opicli1 --id subsys1)
ns0=$(/dpu create frontend nvme namespace --subsystem "$ss0" --volume "$vol0" --id namespace0)
tcp0=$(/dpu create frontend nvme controller tcp --subsystem "$ss0" --ip "127.0.0.1" --port 4420 --id nvmetcp0)
pcie0=$(/dpu create frontend  nvme controller pcie --subsystem "$ss1" --port 0 --pf 0 --vf 0 --id pcie0)
blk0=$(/dpu create frontend virtio blk --volume Malloc1 --pf 0 --vf 0  --id blk0)

sleep 10

/dpu delete frontend virtio blk --name "$blk0"
/dpu delete frontend nvme controller --name "$pcie0"
/dpu delete frontend nvme controller --name "$tcp0"
/dpu delete frontend nvme namespace --name "$ns0"
/dpu delete frontend nvme subsystem --name "$ss1"
/dpu delete frontend nvme subsystem --name "$ss0"
```

Waiting for https://github.com/opiproject/opi-spdk-bridge/pull/837 to pass CI

### CODE REVIEW

File: cmd/storage/frontend/frontend.go [hunk 1]
```diff
@@ -1,49 +1,17 @@
...
 
-	"github.com/spf13/cobra"
-)
-
-const addrCmdLineArg = "addr"
-const timeoutCmdLineArg = "timeout"
-
-// NewStorageCommand tests the storage functionality
-func NewStorageCommand() *cobra.Command {
+// NewCreateCommand creates a new command to delete frontend resources
```
Comment (intelfisz) [inline cmd/storage/frontend/frontend.go:9]: to create

**Review by intelfisz — APPROVED** [review]
lgtm, found just one typo

**Review by sandersms — APPROVED** [review]
LGTM - one failing check to resolve

File: cmd/storage/frontend/frontend.go (hunk 1 above)
Comment (artek-koltun) [reply cmd/storage/frontend/frontend.go]: good catch. Fixed!

**Pull request merged**